components/engine/engine-camel/src/main/java/org/eclipse/dirigible/components/engine/camel/invoke/Invoker.java

```diff
@@ -14,34 +14,76 @@
 import org.eclipse.dirigible.components.engine.camel.processor.CamelProcessor;
 import org.eclipse.dirigible.graalium.core.DirigibleJavascriptCodeRunner;
 import org.eclipse.dirigible.graalium.core.javascript.CalledFromJS;
+import org.graalvm.polyglot.Value;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.io.IOException;
 import java.nio.file.Path;
 import java.util.Map;
 
 @Component
 public class Invoker {
 
+    private DirigibleJavascriptCodeRunner runner;
+
     private final CamelProcessor processor;
 
     @Autowired
     public Invoker(CamelProcessor processor) {
         this.processor = processor;
     }
 
-    public void invoke(Message camelMessage) {
+    public void invoke(Message camelMessage) throws IOException {
+        resetCodeRunner();
         String resourcePath = (String) camelMessage.getExchange()
                                                    .getProperty("resource");
-        String messageBody = camelMessage.getBody(String.class);
-        try (var runner = new DirigibleJavascriptCodeRunner()) {
-            var module = runner.run(Path.of(resourcePath));
-            runner.runMethod(module, "onMessage", messageBody);
+
+        var module = runner.run(Path.of(resourcePath));
+        var result = runner.runMethod(module, "onMessage", wrapCamelMessage(camelMessage));
+
+        camelMessage.getExchange()
+                    .setMessage(unwrapCamelMessage(result));
+    }
+
+    private IntegrationMessage wrapCamelMessage(Message camelMessage) {
+        return new IntegrationMessage(camelMessage);
+    }
+
+    private Message unwrapCamelMessage(Value value) {
+        if (value == null || !value.isHostObject()) {
+            throw new IllegalStateException("Unexpected null value or return type received from " + "@dirigible/integrations::onMessage(). "
+                    + "Expected return type: org.eclipse.dirigible.components.engine.camel.invoke.IntegrationMessage");
+        }
+
+        Object hostObject = value.asHostObject();
+
+        if (!(hostObject instanceof IntegrationMessage integrationMessage)) {
+            throw new IllegalArgumentException("Unexpected return type received from " + "@dirigible/integrations::onMessage(). "
+                    + "Expected return type: org.eclipse.dirigible.components.engine.camel.invoke.IntegrationMessage");
```

Comment:
Expected return type should probably be just IntegrationMessage as this is the TypeScript interface name. Users wouldn't use the Java type I guess

Reply:
Changed to IntegrationMessage